src/file/tokio_compat_file.rs

```diff
@@ -516,6 +539,8 @@ impl AsyncWrite for TokioCompatFile {
             return Poll::Ready(Err(sftp_to_io_error(cancel_error())));
         }
 
+        *this.write_len = 0;
```

Comment:
It's too early to do this, `write_futures` can still return `Poll::Pending`.

Better do this when polling `write_futures`.
I think we also need to embed `usize` of the write into `write_futures`.